trilitech/jstz
# Node & API : Filter logs from `kernel.log` by address

# Context



[Filter kernel.log by address](https://app.asana.com/0/1205770721173531/1205770721347931/f)

# Description

This PR enables the jstz_node endpoint `logs/smart_function/stream` to stream the log relevant to the provided address of the smart function. the endpoint will stream logs in the a json format. 
Example:
```
data: {"contract_address":{"Tz4":"tz4EWCakzUFzj5gHsZZ7dWAkHANxLXTBdpwf"},"request_id":"b9ca156c0ddea66dd8848c5ae298ddf2ac31631eb61f3d48e797bc7d060aa8c7","level":"log","text":"log"}
```

In repl, we keep the pretty log style with the unicode prefix.
 
# Manually testing the PR

1. run the sandbox
2. run the node
3. deploy examples/logs.js 

4. curl http://localhost:8933/logs/SMART_FUNCTION/stream

Additionally, to test that the disconnected clients properly get removed, i printed out the clients address with the number of responsive senders. we can see that once the clients are disconnected, the entry clients in the broadcast map is removed properly after a few ping.


https://github.com/trilitech/jstz/assets/128799322/1dba2b68-96fe-45ec-b4fc-4ee0321ffea2
